gremlin-go/driver/graphTraversalSource.go

```diff
@@ -0,0 +1,23 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package gremlingo
+
+type GraphTraversalSource interface {
```

Comment:
this and other data model  objects associated with Graph should perhaps be separated from connection related stuff as driver/structure/

Reply:
@divijvaidya 

> Thank you for taking up this body of work.
> 
> Please add a style guide for GoLang that other future develops contributing to this code would follow.
> 
> I added very initial comments based on first look. I will delve deeper into this PR in the coming days but as a general rule of thumb we would try to apply the learnings from structure of other clients and improve this one. Primary focus will be testability and debuggability of the client.

Thanks for reviewing Divij, we will definitely look into that. We are having a bit of a hard time with the initial structure because of the way Golang works - if we move a file to a subfolder, it can now only access public facing structs of other classes within the project. However as we get more components in, I think we can begin to organize this better. We look forward to the feedback you will be giving and will incorporate it into the coming Milestones.